x-pack/plugins/endpoint/server/routes/resolver/queries/children.ts

```diff
@@ -3,72 +3,92 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { SearchResponse } from 'elasticsearch';
+import { ResolverEvent } from '../../../../common/types';
 import { ResolverQuery } from './base';
+import { PaginationBuilder, PaginatedResults } from '../utils/pagination';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
 
-export class ChildrenQuery extends ResolverQuery {
-  protected legacyQuery(endpointID: string, uniquePIDs: string[], index: string) {
-    const paginator = this.paginateBy('endgame.serial_event_id', 'endgame.unique_ppid');
+/**
+ * Builds a query for retrieving descendants of a node.
+ */
+export class ChildrenQuery extends ResolverQuery<PaginatedResults> {
+  constructor(
+    private readonly pagination: PaginationBuilder,
+    indexPattern: string,
+    endpointID?: string
+  ) {
+    super(indexPattern, endpointID);
+  }
+
+  protected legacyQuery(endpointID: string, uniquePIDs: string[]): JsonObject {
     return {
-      body: paginator({
-        query: {
-          bool: {
-            filter: [
-              {
-                terms: { 'endgame.unique_ppid': uniquePIDs },
-              },
-              {
-                term: { 'agent.id': endpointID },
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: { 'endgame.unique_ppid': uniquePIDs },
+            },
+            {
+              term: { 'agent.id': endpointID },
+            },
+            {
+              term: { 'event.category': 'process' },
+            },
+            {
+              term: { 'event.kind': 'event' },
+            },
+            {
+              bool: {
+                should: [
+                  {
+                    term: { 'event.type': 'process_start' },
+                  },
+                  {
+                    term: { 'event.action': 'fork_event' },
+                  },
+                ],
               },
-              {
-                term: { 'event.category': 'process' },
-              },
-              {
-                term: { 'event.kind': 'event' },
-              },
-              {
-                bool: {
-                  should: [
-                    {
-                      term: { 'event.type': 'process_start' },
-                    },
-                    {
-                      term: { 'event.action': 'fork_event' },
-                    },
-                  ],
-                },
-              },
-            ],
-          },
+            },
+          ],
         },
-      }),
-      index,
+      },
+      ...this.pagination.buildQueryFields(
```

Comment:
:grey_question: I'm a little confused about how this works... will read into it a little more